near-sdk/src/json_types/public_key.rs

```diff
@@ -51,7 +61,10 @@ impl std::str::FromStr for CurveType {
 ///             .unwrap();
 /// ```
 #[derive(Debug, Clone, PartialEq, PartialOrd, Ord, Eq, BorshDeserialize, BorshSerialize)]
-pub struct Base58PublicKey(pub Vec<u8>);
+pub struct Base58PublicKey {
+    curve: CurveType,
+    data: Vec<u8>,
+}
```

Comment:
extension from previous, this is actually a breaking change of borsh format, so you need to implement borsh (de)serialization methods manually